ui/pages/index.js

```diff
@@ -51,7 +52,9 @@ class Index extends PureComponent {
                   <LegacyI18nProvider>
                     <CurrencyRateProvider>
                       <MetamaskNotificationsProvider>
-                        <Routes />
+                        <TokenDetectionProvider>
```

Comment:
inject `detectedTokensAllChains` as provider , this will add to our global state the object below:
```
{ '0x1': { detected token ...} , ....}
```